### ISSUE TITLE
Usage of alternative (JSON / De-)Serializers

### ISSUE THREAD

**Issue by joergbattermann** [reactiveui/refit#106]
Good morning,

I just started playing with refit and I was wondering whether it was possible to use an alternative Json serializer and deserializer than Json.Net? As far as I can see it its hard-wired mostly in [RequestBuilderImplementation.cs](https://github.com/paulcbetts/refit/blob/master/Refit/RequestBuilderImplementation.cs), but I've come to love [Jil](https://github.com/kevin-montrose/Jil), been using [NetJson](https://github.com/rpgmaker/NetJSON) in another project and some might use [protobuf](https://code.google.com/p/protobuf-net/), [MessagePack](http://msgpack.org/) or some other formats.

While those might not provide support for the exact same amount of platforms as refit does, it might be nice to have a choice and I was wondering, whether the hard-coded usage of Json.net could be de-coupled and one could provide custom (de)serialization providers for refit?

Basically something like providing a Serializer registration point in [https://github.com/paulcbetts/refit/blob/master/Refit/RefitSettings.cs](RefitSettings.cs), use that one instead of Json.Net directly?

I've forked away refit and started dabbling with this idea but wanted to check upfront whether this is something you might be ok with?

Cheers,
-J
